pyccel/ast/core.py

```diff
@@ -211,6 +211,7 @@ def broadcast(shape_1, shape_2):
         new_shape_1 = shape_1
 
     new_shape = []
+    __slots__ = []
```

Comment:
I don't think `__slots__` make sense in functions.
See https://book.pythontips.com/en/latest/__slots__magic.html
for a description of slots

Reply:
Another good reference is here: https://stackoverflow.com/questions/472000/usage-of-slots